WalletWasabi.Fluent/Rpc/WasabiJsonRpcService.cs

```diff
@@ -55,7 +55,10 @@ public object[] GetCoinList()
 
 		AssertWalletIsLoaded();
 		var serverTipHeight = activeWallet.BitcoinStore.SmartHeaderChain.ServerTipHeight;
-		var coinRegistry = activeWallet.Coins as CoinsRegistry;
+		if (activeWallet.Coins is not CoinsRegistry coinRegistry)
```

Comment:
we should throw here - that would be an obvious bug in Coins that would not be CoinsRegistry